conda_build/index.py

```diff
@@ -332,7 +332,7 @@ def _delegated_update_index(
 # Everything below is deprecated to maintain API/feature compatibility.
 
 
-@deprecated("3.25.0", "4.0.0", addendum="Use standalone conda-index.")
+@deprecated("3.25.0", "24.1.0", addendum="Use standalone conda-index.")
```

Comment:
This entire file needs to be deleted ASAP

Reply:
Have we finished rewriting `conda index` so that it calls new conda index, or can we remove the command entirely and require the nicer python3 -m conda_index click CLI

Reply:
the subcommand is marked for deprecation in 24.1.0:

https://github.com/conda/conda-build/blob/328f7e1f9939125efcc934eb5f2c44343b303208/conda_build/cli/main_index.py#L100-L121

it looks like you've marked all of the old index codes for deprecation, we can focus on removing all of this once we finish the November release